### PR TITLE
feat: add noarch python migrator

### PR DESCRIPTION
**Pull request by beckermr**
#### Description:

This PR adds a `noarch: python` migrator per CFEP-25.

#### Checklist:

- [x] Pydantic model updated or no update needed

### CODE REVIEW

File: conda_forge_tick/migrators/noarch_python_min.py [hunk 1]
```diff
@@ -0,0 +1,381 @@
...
+
+
+@functools.lru_cache(maxsize=1)
+def _get_curr_python_min():
+    with pushd(os.environ["CONDA_PREFIX"]):
+        pinnings = parse_config_file(
+            "conda_build_config.yaml",
+            config=Config(),
+        )
+    return pinnings.get("python_min", None)[0]
```
Comment (beckermr) [inline conda_forge_tick/migrators/noarch_python_min.py:31]:
```suggestion
    return (pinnings.get("python_min", [None]) or [None])[0]
```

File: tests/test_noarch_python_min_migrator.py [hunk 2]
```diff
@@ -0,0 +1,541 @@
...
+                """\
+                build:
+                  noarch: python
+
+                requirements:
+                  host:
+                    - python 3.6.*
+                    - numpy
+                  run:
+                    - python >={{ python_min }}
```
Comment (isuruf) [inline tests/test_noarch_python_min_migrator.py:178]: I don't understand this one.

Comment (beckermr) [reply tests/test_noarch_python_min_migrator.py:178]: What precisely do you not understand?

Comment (beckermr) [reply tests/test_noarch_python_min_migrator.py]: fixed

File: conda_forge_tick/make_migrators.py [hunk 3]
```diff
@@ -718,6 +719,14 @@ def initialize_migrators(
         "The package 'mpir' is deprecated and unmaintained. Use 'gmp' instead.",
     )
 
+    with fold_log_lines("making `noarch: python` migrator"):
+        migrators.append(
+            NoarchPythonMinMigrator(
+                graph=gx,
+                pr_limit=PR_LIMIT,
```
Comment (beckermr) [inline conda_forge_tick/make_migrators.py:726]:
```suggestion
                pr_limit=1,  # will turn up later
```

**Pull request merged**